### PR TITLE
typing:/tests/api/test_h-p-s

### PR DESCRIPTION
**Pull request by azerv1**
related #842

### CODE REVIEW

File: rotkehlchen/tests/api/test_settings.py [hunk 1]
```diff
@@ -173,7 +181,7 @@ def test_set_settings(rotkehlchen_api_server):
     for setting, raw_value in original_settings.items():
         if setting in unmodifiable_settings:
             continue
-
+        value: Any = None
```
Comment (LefterisJP) [inline rotkehlchen/tests/api/test_settings.py:184]: Can't you specify the possible values here etc.? Using `Any` in typing is discouraged as it's basically disabling it

```suggestion
        value: str | int | None = None
```

File: rotkehlchen/tests/api/test_snapshots.py [hunk 2]
```diff
@@ -430,7 +438,9 @@ def test_export_snapshot(rotkehlchen_api_server, tmpdir_factory):
     with db.user_write() as cursor:
         _populate_db_with_balances(cursor, db, ts)
         _populate_db_with_location_data(cursor, db, ts)
-        db.set_settings(cursor, ModifiableDBSettings(main_currency=A_EUR))
+        db.set_settings(cursor, ModifiableDBSettings(
+            main_currency=A_EUR.resolve_to_asset_with_oracles()),
+            )
```
Comment (LefterisJP) [inline rotkehlchen/tests/api/test_snapshots.py:443]: This should be aligned with the line where it opens.

```suggestion
        )
```

File: rotkehlchen/tests/api/test_snapshots.py [hunk 3]
```diff
@@ -448,7 +458,9 @@ def test_export_snapshot(rotkehlchen_api_server, tmpdir_factory):
             timestamp_validation_data=(ts, display_date_in_localtime),
         )
 
-        db.set_settings(cursor, ModifiableDBSettings(main_currency=A_ETH))
+        db.set_settings(cursor, ModifiableDBSettings(
+            main_currency=A_ETH.resolve_to_asset_with_oracles()),
+            )
```
Comment (LefterisJP) [inline rotkehlchen/tests/api/test_snapshots.py:463]: same here

```suggestion
        )
```

File: rotkehlchen/tests/api/test_snapshots.py [hunk 4]
```diff
@@ -466,7 +478,9 @@ def test_export_snapshot(rotkehlchen_api_server, tmpdir_factory):
             timestamp_validation_data=(ts, display_date_in_localtime),
         )
 
-        db.set_settings(cursor, ModifiableDBSettings(main_currency=A_USD))
+        db.set_settings(cursor, ModifiableDBSettings(
+            main_currency=A_USD.resolve_to_asset_with_oracles()),
+            )
```
Comment (LefterisJP) [inline rotkehlchen/tests/api/test_snapshots.py:483]:
```suggestion
        )
```

File: rotkehlchen/tests/api/test_snapshots.py [hunk 5]
```diff
@@ -482,14 +496,19 @@ def test_export_snapshot(rotkehlchen_api_server, tmpdir_factory):
...
     db = rotkehlchen_api_server.rest_api.rotkehlchen.data.db
     ts = ts_now()
     csv_dir = str(tmpdir_factory.mktemp('test_csv_dir'))
     with db.user_write() as cursor:
         _populate_db_with_balances_unknown_asset(cursor, ts)
         _populate_db_with_location_data(cursor, db, ts)
-        db.set_settings(cursor, ModifiableDBSettings(main_currency=A_EUR))
+        db.set_settings(cursor, ModifiableDBSettings(
+            main_currency=A_EUR.resolve_to_asset_with_oracles()),
+            )
```
Comment (LefterisJP) [inline rotkehlchen/tests/api/test_snapshots.py:511]:
```suggestion
        )
```

File: rotkehlchen/tests/api/test_snapshots.py [hunk 6]
```diff
@@ -512,13 +531,15 @@ def test_export_snapshot_unknown_asset(rotkehlchen_api_server, tmpdir_factory):
...
+def test_download_snapshot(rotkehlchen_api_server: 'APIServer') -> None:
     db = rotkehlchen_api_server.rest_api.rotkehlchen.data.db
     ts = ts_now()
     with db.user_write() as cursor:
         _populate_db_with_balances(cursor, db, ts)
         _populate_db_with_location_data(cursor, db, ts)
-        db.set_settings(cursor, ModifiableDBSettings(main_currency=A_EUR))
+        db.set_settings(cursor, ModifiableDBSettings(
+            main_currency=A_EUR.resolve_to_asset_with_oracles()),
+            )
```
Comment (LefterisJP) [inline rotkehlchen/tests/api/test_snapshots.py:542]:
```suggestion
        )
```

File: rotkehlchen/tests/api/test_snapshots.py [hunk 7]
```diff
@@ -536,20 +557,25 @@ def test_download_snapshot(rotkehlchen_api_server):
...
+    ) -> None:
     db = rotkehlchen_api_server.rest_api.rotkehlchen.data.db
     ts = ts_now()
     with db.user_write() as cursor:
         _populate_db_with_balances(cursor, db, ts)
         _populate_db_with_location_data(cursor, db, ts)
-        db.set_settings(cursor, ModifiableDBSettings(main_currency=A_EUR))
+        db.set_settings(cursor, ModifiableDBSettings(
+            main_currency=A_EUR.resolve_to_asset_with_oracles()),
+            )
```
Comment (LefterisJP) [inline rotkehlchen/tests/api/test_snapshots.py:578]:
```suggestion
        )
```

File: rotkehlchen/tests/api/test_snapshots.py [hunk 8]
```diff
@@ -674,13 +700,15 @@ def test_import_snapshot(rotkehlchen_api_server, tmpdir_factory):
...
+def test_delete_snapshot(rotkehlchen_api_server: 'APIServer') -> None:
     db = rotkehlchen_api_server.rest_api.rotkehlchen.data.db
     ts = ts_now()
     with db.user_write() as cursor:
         _populate_db_with_balances(cursor, db, ts)
         _populate_db_with_location_data(cursor, db, ts)
-        db.set_settings(cursor, ModifiableDBSettings(main_currency=A_EUR))
+        db.set_settings(cursor, ModifiableDBSettings(
+            main_currency=A_EUR.resolve_to_asset_with_oracles()),
+            )
```
Comment (LefterisJP) [inline rotkehlchen/tests/api/test_snapshots.py:711]:
```suggestion
        )
```

**Review by LefterisJP — COMMENTED** [review]
Left some things to change

**Review by yabirgb — APPROVED** [review]
lgtm

**Pull request merged**